src/main/java/pingis/controllers/LiveChallengeController.java

```diff
@@ -86,26 +95,111 @@ public RedirectView createTaskPair(String testTaskName, String implementationTas
-    
     Challenge currentChallenge = challengeService.findOne(challengeId);
-    logger.info("Current Challenge fetched: " + currentChallenge);
-    if (!currentChallenge.getIsOpen()) {
-      logger.info("Trying to play a closed challenge. Redirecting to /error.");
+    if (!gameplayService.isParticipating(currentChallenge)) {
+      logger.info("User trying to close somebody else's challenge. Redirecting to /error.");
+      redirectAttributes.addFlashAttribute("message", "user not in challenge");
       return new RedirectView("/error");
     }
-    
+
+    currentChallenge.setOpen(false);
+    challengeService.save(currentChallenge);
+    redirectAttributes.addFlashAttribute("message", "Challenge closed.");
+    logger.info("Closing challenge: " + currentChallenge.getId());
+    return new RedirectView("/user");
+  }
+
+  @RequestMapping("/newArcadeChallenge")
+  public String newArcadeChallenge(Model model) {
+    return "newarcadechallenge";
+  }
+
+  @RequestMapping(value = "/createArcadeChallenge", method = RequestMethod.POST)
+  public RedirectView createArcadeChallenge(RedirectAttributes redirectAttributes,
+      String challengeName,
+      String challengeDesc,
+      String realm,
+      int level) {
+
+    Challenge arcadeChallenge = new Challenge(
+        challengeName,
+        userService.getCurrentUser(),
+        challengeDesc,
+        ChallengeType.ARCADE);
+    logger.info("Created new arcade challenge: " + arcadeChallenge.toString());
+
+    try {
+      arcadeChallenge.setRealm(Realm.valueOf(realm.toUpperCase()));
+    } catch (Exception e) {
+      logger.info("Realm {} does not exist. Redirecting to /error.", realm);
+      return new RedirectView("/error");
+    }
+    arcadeChallenge.setOpen(true);
+    arcadeChallenge.setLevel(level);
+    challengeService.save(arcadeChallenge);
+    return new RedirectView("/user");
+  }
+
+  @RequestMapping("/newArcadeSession")
+  public RedirectView newArcadeSession(RedirectAttributes redirectAttributes,
+      @RequestParam String realm) {
+    logger.info("playArcade method entered");
+    User player = userService.getCurrentUser();
+    Realm currentRealm = null;
+    try {
+      logger.info("Trying to get realm: {}", realm);
+      currentRealm = Realm.valueOf(realm.toUpperCase());
+    } catch (Exception e) {
+      logger.info("Realm {} does not exist. Redirecting to /error.", realm);
+      return new RedirectView("/error");
+    }
+    Challenge challenge = gameplayService.getArcadeChallenge(currentRealm);
+    if (player.getMostRecentArcadeInstance() != null
+        && !(player.getMostRecentArcadeInstance().getChallenge().getRealm() == currentRealm)) {
+      List<TaskInstance> instances = taskInstanceService.getAllByChallenge(challenge);
+      Optional<TaskInstance> unfinished = instances.stream()
+          .filter(i -> i.getUser().equals(player))
+          .filter(i -> i.getStatus() == CodeStatus.IN_PROGRESS)
+          .findFirst();
```

Comment:
Might be nice as a method of `TaskInstanceService`, especially since the `newArcadeSession` is somewhat long and has multiple if statements.